migration_tools/src/extract_dataset.py

```diff
@@ -123,7 +170,9 @@ def extract_tables(
                 job_config=EXTRACT_JOB_CONFIG,
             )
         )
-        logger.info(f"Table {table} extracted to {gcs_path}.")
+        if len(jobs) >= limitation:
+            raise ExtractQuotaExceeded()
```

Comment:
I think this exception should be raised before creating the job that would exceed the limit.  I'd move this to line 162 to be the first thing executed inside the for loop.